addons/website/static/tests/tours/s_text_image_options.js

```diff
@@ -0,0 +1,25 @@
+/** @odoo-module */
+
+import wTourUtils from 'website.tour_utils';
+
+wTourUtils.registerEditionTour('s_text_image_options', {
+    test: true,
+    url: '/',
+    edition: true,
+}, [
+wTourUtils.dragNDrop({
+    id: 's_text_image',
+    name: 'Text - Image',
+}), {
+    content: "Click on the first paragraph.",
+    trigger: 'iframe .s_text_image p'
+}, {
+    content: "The toolbar should be visible. Click on the width option.",
+    trigger: '[data-select-class="o_container_small"]',
+    extra_trigger: '#oe_snippets .o_we_customize_panel > #o_we_editor_toolbar_container',
+}, {
+    content: "The snippet should have the correct class. The toolbar should be hidden.",
+    trigger: 'iframe .s_text_image > .o_container_small',
+    extra_trigger: '#oe_snippets .o_we_customize_panel:not(> #o_we_editor_toolbar_container:visible)',
```

Comment:
You did not test your test here: your test always passes without your fix, so it does not test anything :/ `.o_we_customize_panel:not(> #o_we_editor_toolbar_container:visible)`, that's definitely not a valid selector, jQuery probably just ignores the `:not` because he don't know what you want from that... maybe you meant `:not(:has(` ? :/ But that would be too fragile as if the XML structure changes, the `:not(:has(` selector would always match and thus you are back not testing anything. Instead, here your extra_trigger should be your current trigger, the trigger should be about finding the customize_panel, then in the run method you can use `this.$anchor` (if you don't define the run function as an arrow function of course) and make a proper check about the text tools visibility.

I won't review further the first commit, make sure everything's ok before pinging me back :grimacing: